packages/search-in-workspace/src/browser/components/input-with-history.tsx

```diff
@@ -0,0 +1,141 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from '@theia/core/shared/react';
+import { Key, KeyCode } from '@theia/core/lib/browser';
+import debounce = require('@theia/core/shared/lodash.debounce');
+
+interface HistoryState {
+    history: string[];
+    index: number;
+};
+type InputAttributes = React.InputHTMLAttributes<HTMLInputElement>;
+
+export class InputWithHistory extends React.Component<InputAttributes, HistoryState> {
```

Comment:
I believe we should prefix the file and class with `search` or `search-in-workspace`?

- https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#unique-names

It follows the `components/` for `preferences` as well:
- https://github.com/eclipse-theia/theia/tree/master/packages/preferences/src/browser/views/components

Reply:
If the `InputWithHistory` is not specific to `search-in-workspace` perhaps it can be added to `@theia/core` and potentially re-used across the framework if needed?

Reply:
I've added the prefixes.

To be suitable for re-use across the framework, I think it would need to handle its own persistence, and so would need to be a fuller custom class rather than a simple React component.